### PR TITLE
Upgrade ESLint

### LINKED ISSUE #1297: Upgrade ESLint to version 8
https://eslint.org/docs/8.0.0/user-guide/migrating-to-8.0.0

### PR DESCRIPTION
**Pull request by samrobbins85**
This upgrades ESLint to version 8, in the process upgrading from [babel-eslint](https://github.com/babel/babel-eslint) to [@babel/eslint-parser](https://github.com/babel/babel/tree/main/eslint/babel-eslint-parser) as babel-eslint has been deprecated and isn't compatible with ESLint 8

Closes #1297

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,10 +1,17 @@
 module.exports = {
-  parser: 'babel-eslint',
+  parser: '@babel/eslint-parser',
+  parserOptions: {
+    requireConfigFile: false,
+    babelOptions: {
+      presets: ['@babel/preset-react'],
```
Comment (mfix22) [inline .eslintrc.js:6]: I am pretty sure we want to just use `next/babel` (or something) here

File: package.json [hunk 2]
```diff
@@ -68,10 +68,14 @@
   },
   "devDependencies": {
     "@applitools/eyes-cypress": "^3.23.4",
+    "@babel/core": "^7.16.5",
```
Comment (mfix22) [inline package.json:71]: All the required babel tools should be already included with `next` (except `@babel/eslint-parser` maybe)

File: package.json [hunk 3]
```diff
@@ -68,10 +68,14 @@
...
+    "@babel/eslint-parser": "^7.16.5",
+    "@babel/plugin-proposal-class-properties": "^7.16.5",
+    "@babel/preset-react": "^7.16.5",
     "all-contributors-cli": "^6.20.0",
     "babel-eslint": "^10.1.0",
```
Comment (mfix22) [inline package.json:76]: Please remove unused dependency.

**Review by mfix22 — CHANGES_REQUESTED** [review]
Thanks for taking care of this @samrobbins85! Couple tweaks and this should be g2g.

File: .eslintrc.js (hunk 1 above)
Comment (mfix22) [reply .eslintrc.js:6]:
```
"presets": ["next/babel"]
```
https://nextjs.org/docs/advanced-features/customizing-babel-config

Comment (samrobbins85) [reply .eslintrc.js]: Didn't realize Next had a babel config, thanks!

File: package.json (hunk 2 above)
Comment (samrobbins85) [reply package.json]: Needed to keep the parser, but everything else removed

File: package.json (hunk 3 above)
Comment (samrobbins85) [reply package.json:73]: Sorry, I forgot I left that one in

**Comment by mfix22** [comment]
Thanks @samrobbins85!

**Pull request merged**